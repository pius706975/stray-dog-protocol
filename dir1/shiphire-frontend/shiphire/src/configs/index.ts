export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"