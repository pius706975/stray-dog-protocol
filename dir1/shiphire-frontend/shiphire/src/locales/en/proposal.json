{
    "labelButtonNegotiate": "Negotiate",
    "labelButtonAccept": "Accept Proposal"
}
