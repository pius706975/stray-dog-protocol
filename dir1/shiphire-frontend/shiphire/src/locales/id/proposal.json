{
    "labelButtonNegotiate": "Negosiasi",
    "labelButtonAccept": "Terima Proposal"
}
