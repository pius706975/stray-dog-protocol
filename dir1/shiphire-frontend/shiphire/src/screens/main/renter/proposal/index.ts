export { default as ProposalPreview } from "./ProposalPreview"