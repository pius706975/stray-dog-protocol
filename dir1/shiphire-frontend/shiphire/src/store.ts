import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
