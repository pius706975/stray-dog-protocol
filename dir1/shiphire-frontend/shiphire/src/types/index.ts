export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'