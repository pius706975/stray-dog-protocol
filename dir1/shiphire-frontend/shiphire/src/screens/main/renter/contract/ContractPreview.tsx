import React from 'react';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    CheckIcon,
    Color,
    EditIcon,
    USERDATA,
    getDataFromLocalStorage,
} from '../../../../configs';
import { useSendOTPSignContract } from '../../../../hooks';
import { progressIndicatorSlice } from '../../../../slices';
import { ContractPreviewProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';
import { OTPModal } from './component';
import { useTranslation } from 'react-i18next';

const ContractPreview: React.FC<ContractPreviewProps> = props => {
    const dispatch = useDispatch();
    const { contractUrl, rentalId } = props.route.params;
    const { t } = useTranslation('common');
    const mutationSendOTP = useSendOTPSignContract();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const [visible, setVisible] = React.useState<boolean>(false);
    const [renterEmail, setRenterEmail] = React.useState<string>('');
    const [contractLoading, setContractLoading] =
        React.useState<boolean>(false);

    const handleAcceptPressed = () => {
        mutationSendOTP.mutate(undefined, {
            onSuccess: () => {
                setVisible(!visible);
            },
            onError: err => {
                handleAxiosError(err);
            },
        });
    };

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(resp => {
            setRenterEmail(resp.email);
        });
        setContractLoading(true);
    }, []);

    React.useEffect(() => {
        dispatch(showProgressIndicator());
        setTimeout(() => {
            dispatch(hideProgressIndicator());
        }, 1000);
    }, [contractLoading]);

    return (
        <ScreenLayout
            testId="ContractPreviewScreen"
            backgroundColor="light"
            padding={10}>
            {contractUrl && (
                <PDFView
                    fadeInDuration={250.0}
                    style={{
                        flex: 1,
                        borderWidth: 1,
                        backgroundColor: Color.bgColor,
                    }}
                    resource={contractUrl}
                    resourceType={'url'}
                    onLoad={() => {
                        setContractLoading(false);
                        setTimeout(() => {
                            dispatch(hideProgressIndicator());
                        }, 1000);
                    }}
                    onError={error => console.log('Cannot render PDF', error)}
                />
            )}
            <View
                row
                spread
                center
                flex
                style={{
                    position: 'absolute',
                    bottom: 30,
                }}>
                {/* <View flex row center>
                    <Button
                        title={t('ContractPreview.labelButtonNegotiate')}
                        onSubmit={() => {}}
                        color="warning"
                        leftIcon={<EditIcon />}
                    />
                </View> */}
                <View flex row right>
                    <Button
                        testID="acceptButton"
                        title={t('ContractPreview.labelButtonAcceptContract')}
                        onSubmit={handleAcceptPressed}
                        color="success"
                        leftIcon={<CheckIcon />}
                        isSubmitting={mutationSendOTP.isLoading}
                    />
                </View>
            </View>
            {visible && (
                <OTPModal
                    visible={visible}
                    setVisible={setVisible}
                    parentProp={props}
                    rentalId={rentalId}
                    renterEmail={renterEmail}
                />
            )}
        </ScreenLayout>
    );
};

export default ContractPreview;
