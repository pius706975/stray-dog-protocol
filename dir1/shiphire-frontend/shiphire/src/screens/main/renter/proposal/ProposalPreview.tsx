import React from 'react';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    CheckIcon,
    Color,
    EditIcon,
    USERDATA,
    getDataFromLocalStorage,
} from '../../../../configs';
import { useSendOTPSignProposal } from '../../../../hooks';
import { progressIndicatorSlice } from '../../../../slices';
import { ProposalPreviewProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';
import { OTPModal } from './component';
import { useTranslation } from 'react-i18next';

const ProposalPreview: React.FC<ProposalPreviewProps> = props => {
    const dispatch = useDispatch();
    const { proposalUrl, rentalId } = props.route.params;
    const mutationSendOTP = useSendOTPSignProposal();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const [visible, setVisible] = React.useState<boolean>(false);
    const [renterEmail, setRenterEmail] = React.useState<string>('');
    const [proposalLoading, setProposalLoading] =
        React.useState<boolean>(false);
    const { t } = useTranslation('proposal');

    const handleAcceptPressed = () => {
        mutationSendOTP.mutate(undefined, {
            onSuccess: () => {
                setVisible(!visible);
            },
            onError: err => {
                handleAxiosError(err);
            },
        });
    };

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(resp => {
            setRenterEmail(resp.email);
        });
        setProposalLoading(true);
    }, []);

    React.useEffect(() => {
        dispatch(showProgressIndicator());
        setTimeout(() => {
            dispatch(hideProgressIndicator());
        }, 1000);
    }, [proposalLoading]);

    return (
        <ScreenLayout
            testId="ProposalPreviewScreen"
            backgroundColor="light"
            padding={10}>
            {proposalUrl && (
                <PDFView
                    fadeInDuration={250.0}
                    style={{
                        flex: 1,
                        borderWidth: 1,
                        backgroundColor: Color.bgColor,
                    }}
                    resource={proposalUrl}
                    resourceType={'url'}
                    onLoad={() => {
                        setProposalLoading(false);
                        setTimeout(() => {
                            dispatch(hideProgressIndicator());
                        }, 1000);
                    }}
                    onError={error => console.log('Cannot render PDF', error)}
                />
            )}
            <View
                row
                spread
                center
                flex
                style={{
                    position: 'absolute',
                    bottom: 30,
                }}>
                <View flex row center>
                    <Button
                        title={t('labelButtonNegotiate')}
                        onSubmit={() => {}}
                        color="warning"
                        leftIcon={<EditIcon />}
                    />
                </View>
                <View flex row center>
                    <Button
                        testID="acceptButton"
                        title={t('labelButtonAccept')}
                        onSubmit={handleAcceptPressed}
                        color="success"
                        leftIcon={<CheckIcon />}
                        isSubmitting={mutationSendOTP.isLoading}
                    />
                </View>
            </View>
            {visible && (
                <OTPModal
                    visible={visible}
                    setVisible={setVisible}
                    parentProp={props}
                    rentalId={rentalId}
                    renterEmail={renterEmail}
                />
            )}
        </ScreenLayout>
    );
};

export default ProposalPreview;
