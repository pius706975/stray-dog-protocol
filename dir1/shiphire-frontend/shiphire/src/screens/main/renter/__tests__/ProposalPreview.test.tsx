import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import ContractPreview from '../contract/ContractPreview';
import { CloseIcon, TOKEN, USERDATA } from '../../../../configs';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { QueryClient, QueryClientProvider } from 'react-query';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { MainScreenParamList, MainStackParamList } from '../../../../types';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import PDFView from 'react-native-view-pdf';
import { act } from 'react-test-renderer';
import { ProposalPreview } from '../proposal';
import {
    useSendOTPSignProposal,
    useVerifySignProposalOTP,
} from '../../../../hooks';

const queryClient = new QueryClient();
const Tab = createBottomTabNavigator<MainScreenParamList>();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: key => key,
    }),
}));

const mockParams = {
    proposalUrl: 'https://example.com/contract.pdf',
    rentalId: '123',
};

const mockProposalComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="ProposalPreview"
                        component={ProposalPreview}
                        initialParams={mockParams}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

AsyncStorage.setItem(
    USERDATA,
    JSON.stringify({
        name: 'Usmanul',
        email: 'usman@email.com',
        password: 'test',
        phoneNumber: '12313123',
        isVerified: true,
        isCompanySubmitted: true,
    }),
);
AsyncStorage.setItem(
    TOKEN,
    JSON.stringify({
        accessToken: 'test',
        refreshToken: 'test',
    }),
);

describe('Proposal Preview', () => {
    it('should render the screen correctly', async () => {
        const proposalPreview = render(mockProposalComponent()).toJSON();
        expect(proposalPreview).toMatchSnapshot();
    });
});
describe('Receipt Preview', () => {
    it('should render the proposal', async () => {
        render(mockProposalComponent());
        const pdfView = render(
            <PDFView
                resource="https://example.com/contract.pdf"
                resourceType="url"
            />,
        );
        await act(() => {
            expect(pdfView).toBeTruthy();
        });
    });
    it('should not render the PDF view if proposal is not provided', async () => {
        render(mockProposalComponent());
        const pdfView = render(
            <PDFView
                resource={''}
                onError={error => {
                    console.log('Cannot render PDF', error);
                }}
            />,
        );

        const progressState = store.getState().progressIndicator;
        expect(progressState).toEqual({
            visible: true,
        });
        await waitFor(() => {
            const progressState = store.getState().progressIndicator;
            expect(progressState).toEqual({
                visible: false,
            });
        });
        await act(async () => {
            expect(pdfView).toBeDefined();
        });
    });
    it('should call handleAcceptPressed when the accept button is pressed', async () => {
        const { getByTestId } = render(mockProposalComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });
    });
    it('should call OTPModal when the accept button is pressed', async () => {
        const { getByTestId } = render(mockProposalComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        await waitFor(() => {
            expect(modal).toBeTruthy();
        });
    });
    it('should call OTPModal when the accept button is pressed and render the component', async () => {
        const { getByTestId, getByText } = render(mockProposalComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        await waitFor(() => {
            expect(modal).toBeTruthy();
        });

        const pressVisible = getByTestId('pressVisible');

        expect(pressVisible).toBeDefined();

        expect(getByText('OTP Verification')).toBeTruthy();

        const press = getByTestId('pressCount');

        expect(press).toBeDefined();
    });
    it('should call OTPModal when the accept button is pressed and input the field', async () => {
        jest.useFakeTimers();

        const { getByTestId } = render(mockProposalComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        await waitFor(() => {
            expect(modal).toBeTruthy();
        });

        const input = getByTestId('OTPInput');

        fireEvent.changeText(input, '123456');

        expect(input.props.value).toEqual('123456');

        const verifyButton = getByTestId('verifyButton');

        fireEvent.press(verifyButton);

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/verify-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });
    });
});
