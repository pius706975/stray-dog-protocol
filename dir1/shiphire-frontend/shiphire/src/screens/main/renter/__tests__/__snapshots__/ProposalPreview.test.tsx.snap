// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposal Preview should render the screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="ProposalPreviewScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  center={true}
                  row={true}
                  spread={true}
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "bottom": 30,
                        "position": "absolute",
                      },
                    ]
                  }
                >
                  <View
                    center={true}
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                        },
                        undefined,
                        {},
                        {},
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#d9a318",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 10,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="20"
                        bbWidth="20"
                        fill="none"
                        focusable={false}
                        height={20}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "flex": 0,
                              "height": 20,
                              "width": 20,
                            },
                          ]
                        }
                        vbHeight={20}
                        vbWidth={20}
                        width={20}
                        xml="<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_29270)">
    <path d="M2.5 14.5501V17.0835C2.5 17.3168 2.68333 17.5001 2.91667 17.5001H5.45C5.55833 17.5001 5.66667 17.4585 5.74167 17.3751L14.8417 8.28346L11.7167 5.15846L2.625 14.2501C2.54167 14.3335 2.5 14.4335 2.5 14.5501ZM17.2583 5.8668C17.5833 5.5418 17.5833 5.0168 17.2583 4.6918L15.3083 2.7418C14.9833 2.4168 14.4583 2.4168 14.1333 2.7418L12.6083 4.2668L15.7333 7.3918L17.2583 5.8668V5.8668Z" fill=#f0f6fb/>
    </g>
    <defs>
    <clipPath id="clip0_77_29270">
    <rect width="20" height="20" fill="white"/>
    </clipPath>
    </defs>
    </svg>    
        "
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <RNSVGGroup
                          fill={null}
                          propList={
                            [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGGroup
                            clipPath="clip0_77_29270"
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGPath
                              d="M2.5 14.5501V17.0835C2.5 17.3168 2.68333 17.5001 2.91667 17.5001H5.45C5.55833 17.5001 5.66667 17.4585 5.74167 17.3751L14.8417 8.28346L11.7167 5.15846L2.625 14.2501C2.54167 14.3335 2.5 14.4335 2.5 14.5501ZM17.2583 5.8668C17.5833 5.5418 17.5833 5.0168 17.2583 4.6918L15.3083 2.7418C14.9833 2.4168 14.4583 2.4168 14.1333 2.7418L12.6083 4.2668L15.7333 7.3918L17.2583 5.8668V5.8668Z"
                              fill={
                                {
                                  "payload": 4293981947,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                          <RNSVGDefs>
                            <RNSVGClipPath
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              name="clip0_77_29270"
                            >
                              <RNSVGRect
                                fill={
                                  {
                                    "payload": 4294967295,
                                    "type": 0,
                                  }
                                }
                                height="20"
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                                width="20"
                                x="0"
                                y="0"
                              />
                            </RNSVGClipPath>
                          </RNSVGDefs>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        labelButtonNegotiate
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    center={true}
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                        },
                        undefined,
                        {},
                        {},
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#1fcc44",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 10,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="acceptButton"
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="24"
                        bbWidth="24"
                        fill="none"
                        focusable={false}
                        height={24}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "flex": 0,
                              "height": 24,
                              "width": 24,
                            },
                          ]
                        }
                        vbHeight={24}
                        vbWidth={24}
                        width={24}
                        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <g clip-path="url(#clip0_77_30622)">
        <path d="M9.00001 16.1703L5.53001 12.7003C5.14002 12.3103 4.51001 12.3103 4.12001 12.7003C3.73001 13.0903 3.73001 13.7203 4.12001 14.1103L8.30001 18.2903C8.69001 18.6803 9.32001 18.6803 9.71001 18.2903L20.29 7.71031C20.68 7.32031 20.68 6.69031 20.29 6.30031C19.9 5.91031 19.27 5.91031 18.88 6.30031L9.00001 16.1703Z" fill="#f0f6fb"/>
        </g>
        <defs>
        <clipPath id="clip0_77_30622">
        <rect width="24" height="24" fill="white"/>
        </clipPath>
        </defs>
        </svg>
        "
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <RNSVGGroup
                          fill={null}
                          propList={
                            [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGGroup
                            clipPath="clip0_77_30622"
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGPath
                              d="M9.00001 16.1703L5.53001 12.7003C5.14002 12.3103 4.51001 12.3103 4.12001 12.7003C3.73001 13.0903 3.73001 13.7203 4.12001 14.1103L8.30001 18.2903C8.69001 18.6803 9.32001 18.6803 9.71001 18.2903L20.29 7.71031C20.68 7.32031 20.68 6.69031 20.29 6.30031C19.9 5.91031 19.27 5.91031 18.88 6.30031L9.00001 16.1703Z"
                              fill={
                                {
                                  "payload": 4293981947,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                          <RNSVGDefs>
                            <RNSVGClipPath
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              name="clip0_77_30622"
                            >
                              <RNSVGRect
                                fill={
                                  {
                                    "payload": 4294967295,
                                    "type": 0,
                                  }
                                }
                                height="24"
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                                width="24"
                                x="0"
                                y="0"
                              />
                            </RNSVGClipPath>
                          </RNSVGDefs>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        labelButtonAccept
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ProposalPreview"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
